libs/beacon/src/beacon_setup_service.cpp

```diff
@@ -117,6 +145,19 @@ BeaconSetupService::State BeaconSetupService::OnIdle()
     else
     {
       // Initiating setup
+      std::set<MuddleAddress> cabinet;
+      for (auto &m : beacon_->aeon.members)
+      {
+        cabinet.insert(m.identifier());
+      }
+
+      pre_dkg_rbc_.ResetCabinet(cabinet);
```

Comment:
I wonder if not resetting the cabinet here will have problems for people in 'observe' mode, that is, they will still accept and process messages from former cabinet members

Reply:
This is true. I have added resets for the rbc and pre_dkg_rbc with an empty cabinet